homeassistant/components/recorder/core.py

```diff
@@ -418,12 +418,7 @@ def _reached_max_backlog_percentage(self, percentage: int) -> bool:
         # If they have more RAM available, keep filling the backlog
         # since we do not want to stop recording events or give the
         # user a bad backup when they have plenty of RAM available.
-        max_queue_backlog = int(
-            QUEUE_PERCENTAGE_ALLOWED_AVAILABLE_MEMORY
-            * (self._available_memory() / ESTIMATED_QUEUE_ITEM_SIZE)
-        )
-        self.max_backlog = max(max_queue_backlog, MAX_QUEUE_BACKLOG_MIN_VALUE)
-        return current_backlog >= (max_queue_backlog * percentage_modifier)
+        return self._available_memory() < MIN_AVAILABLE_MEMORY_FOR_QUEUE_BACKLOG
```

Comment:
Should the constant be called “max” and not “min”?

Reply:
min seems right here since we want to return `True` as soon as the available memory is less than the minimum allowed because we need to abort when when the backlog gets too big. Maybe I should rename the function

Reply:
Adjusted the function and dropped the percentage since the check runs every 10s and 90% vs 100% is unlikely to be reached in 10s